### PR TITLE
feat(resources): add Severity to detail panel

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

This PR follows the one regarding the listing.
This PR intends to add a new `severity` entry key for host / service detail panel.

Host detail

image

Service Detail

image

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Will be fully tested with the Feature story

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: doc/API/ResourceHostDetail.yaml [hunk 1]
```diff
@@ -151,6 +146,60 @@ Monitoring.HostDetail:
...
+          example: 28
+        name:
+          type: string
+          description: "Name of the Severity"
+          example: "High"
+        type:
+          type: string
+          description: "Type of the Severity"
+          enum: ['host', 'service']
+        level:
```
Comment (adr-mo) [inline doc/API/ResourceHostDetail.yaml:182]: Add min max [0 - 100]

File: doc/API/ResourceHostDetail.yaml [hunk 2]
```diff
@@ -151,6 +146,60 @@ Monitoring.HostDetail:
...
+          type: integer
+          description: "Level of the Severity"
+          example: 50
+        icon:
+          type: object
+          nullable: true
```
Comment (adr-mo) [inline doc/API/ResourceHostDetail.yaml]:
```suggestion
```

File: doc/API/ResourceServiceDetail.yaml [hunk 3]
```diff
@@ -154,6 +148,63 @@ Monitoring.ServiceDetail:
...
+          example: 28
+        name:
+          type: string
+          description: "Name of the Severity"
+          example: "High"
+        type:
+          type: string
+          description: "Type of the Severity"
+          enum: ['host', 'service']
+        level:
```
Comment (adr-mo) [inline doc/API/ResourceServiceDetail.yaml:188]: Same that for host detail add min/max

File: doc/API/centreon-api-v22.10.yaml [hunk 4]
```diff
@@ -1256,7 +1256,6 @@ paths:
         * notes_url
         * parent_name
         * parent_status
-        * severity_level
```
Comment (adr-mo) [inline doc/API/centreon-api-v22.10.yaml:1259]: to check

File: src/Core/Application/RealTime/UseCase/FindHost/FindHost.php [hunk 5]
```diff
@@ -110,6 +114,14 @@ public function __invoke(int $hostId, FindHostPresenterInterface $presenter): vo
 
         $host->setCategories($categories);
 
+        $severity = $this->severityRepository->findByResourceAndTypeId(
```
Comment (adr-mo) [inline src/Core/Application/RealTime/UseCase/FindHost/FindHost.php:125]: move logs at the UseCase level and not repository

File: src/Core/Application/RealTime/UseCase/FindHost/FindHostResponse.php [hunk 6]
```diff
@@ -174,6 +170,11 @@ class FindHostResponse
      */
     public $categories;
 
+    /**
+     * @var array<string, mixed>
+     */
+    public $severity;
```
Comment (adr-mo) [inline src/Core/Application/RealTime/UseCase/FindHost/FindHostResponse.php]:
```suggestion
    public array $severity = [];
```

File: src/Core/Application/RealTime/UseCase/FindHost/FindHostResponse.php [hunk 7]
```diff
@@ -174,6 +170,11 @@ class FindHostResponse
      */
     public $categories;
```
Comment (adr-mo) [inline src/Core/Application/RealTime/UseCase/FindHost/FindHostResponse.php]:
```suggestion
    public array $categories = [];
```

File: src/Core/Application/RealTime/UseCase/FindService/FindServiceResponse.php [hunk 8]
```diff
@@ -175,6 +171,11 @@ class FindServiceResponse
      */
     public $hasGraphData;
 
+    /**
+     * @var array<string, mixed>
+     */
+    public array $severity;
```
Comment (adr-mo) [inline src/Core/Application/RealTime/UseCase/FindService/FindServiceResponse.php]: same as host response

File: src/Core/Application/RealTime/UseCase/FindService/FindServiceResponse.php [hunk 9]
```diff
@@ -206,6 +209,7 @@ public function __construct(
         $this->acknowledgement = $this->acknowledgementToArray($acknowledgement);
         $this->host = $this->hostToArray($host);
         $this->categories = $this->tagsToArray($serviceCategories);
+        $this->severity = $this->severityToArray($severity);
```
Comment (adr-mo) [inline src/Core/Application/RealTime/UseCase/FindService/FindServiceResponse.php]: Do not set an empty array when null but set property to null

File: src/Core/Severity/RealTime/Application/Repository/ReadSeverityRepositoryInterface.php [hunk 10]
```diff
@@ -33,4 +33,14 @@ interface ReadSeverityRepositoryInterface
...
      */
     public function findAllByTypeId(int $typeId): array;
+
+    /**
+     * Finds a Severity by id, parentId and typeId
+     *
+     * @param integer $id
+     * @param integer $parentId
+     * @param integer $typeId
+     * @return Severity|null
```
Comment (adr-mo) [inline src/Core/Severity/RealTime/Application/Repository/ReadSeverityRepositoryInterface.php]:
```suggestion
     * @param integer $resourceId
     * @param integer $resourceParentId
     * @param integer $typeId
     * @return Severity|null
```

File: src/Core/Severity/RealTime/Application/Repository/ReadSeverityRepositoryInterface.php [hunk 11]
```diff
@@ -33,4 +33,14 @@ interface ReadSeverityRepositoryInterface
...
+     */
+    public function findByResourceAndTypeId(int $id, int $parentId, int $typeId): ?Severity;
```
Comment (adr-mo) [inline src/Core/Severity/RealTime/Application/Repository/ReadSeverityRepositoryInterface.php]:
```suggestion
    public function findByResourceAndTypeId(int $resourceId, int $resourceParentId, int $typeId): ?Severity;
```

File: src/Core/Severity/RealTime/Infrastructure/Repository/DbReadSeverityRepository.php [hunk 12]
```diff
@@ -115,4 +121,54 @@ public function findAllByTypeId(int $typeId): array
 
         return $severities;
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function findByResourceAndTypeId(int $id, int $parentId, int $typeId): ?Severity
```
Comment (adr-mo) [inline src/Core/Severity/RealTime/Infrastructure/Repository/DbReadSeverityRepository.php]: make changes according to the interface

File: tests/php/Core/Application/RealTime/UseCase/FindHost/FindHostTest.php [hunk 13]
```diff
@@ -217,6 +231,10 @@
     expect($presenter->response->acknowledgement['host_id'])->toBe($this->acknowledgement->getHostId());
     expect($presenter->response->categories[0]['id'])->toBe($this->category->getId());
     expect($presenter->response->categories[0]['name'])->toBe($this->category->getName());
+    expect($presenter->response->severity['id'])->toBe($this->severity->getId());
+    expect($presenter->response->severity['name'])->toBe($this->severity->getName());
+    expect($presenter->response->severity['type'])->toBe($this->severity->getTypeAsString());
+    expect($presenter->response->severity['level'])->toBe($this->severity->getLevel());
```
Comment (adr-mo) [inline tests/php/Core/Application/RealTime/UseCase/FindHost/FindHostTest.php]: Add check on icon entry

File: tests/php/Core/Application/RealTime/UseCase/FindHost/FindHostTest.php [hunk 14]
```diff
@@ -308,4 +331,8 @@
     expect($presenter->response->acknowledgement['entry_time'])->toBe($this->acknowledgement->getEntryTime());
     expect($presenter->response->categories[0]['id'])->toBe($this->category->getId());
     expect($presenter->response->categories[0]['name'])->toBe($this->category->getName());
+    expect($presenter->response->severity['id'])->toBe($this->severity->getId());
+    expect($presenter->response->severity['name'])->toBe($this->severity->getName());
+    expect($presenter->response->severity['type'])->toBe($this->severity->getTypeAsString());
+    expect($presenter->response->severity['level'])->toBe($this->severity->getLevel());
```
Comment (adr-mo) [inline tests/php/Core/Application/RealTime/UseCase/FindHost/FindHostTest.php]: same as above

File: tests/php/Core/Domain/RealTime/Model/HostTest.php [hunk 15]
```diff
@@ -118,7 +118,6 @@ public static function createHostModel(): Host
             ->setIsAcknowledged(false)
             ->setActiveChecks(true)
             ->setPassiveChecks(false)
-            ->setSeverityLevel(10)
```
Comment (adr-mo) [inline tests/php/Core/Domain/RealTime/Model/HostTest.php:121]: replace by setSeverity

File: tests/php/Core/Domain/RealTime/Model/ServiceTest.php [hunk 16]
```diff
@@ -83,7 +83,6 @@ public static function createServiceModel(): Service
             ->setIsAcknowledged(false)
             ->setActiveChecks(true)
             ->setPassiveChecks(false)
-            ->setSeverityLevel(10)
```
Comment (adr-mo) [inline tests/php/Core/Domain/RealTime/Model/ServiceTest.php:86]: replace by setSeverity

Comment (adr-mo) [reply tests/php/Core/Domain/RealTime/Model/ServiceTest.php:86]: Not done finally

**Review by jeremyjaouen — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**